app/producers/application_producer.rb

```diff
@@ -26,8 +28,19 @@ def logger
       Rails.logger
     end
 
+    def kafka_ca_cert
+      File.read(self::SSL_CA_LOCATION) if self::SECURITY_PROTOCOL == 'ssl'
+    end
+
+    def kafka_config
+      {}.tap do |config|
+        config[:client_id] = self::CLIENT_ID
+        config[:ssl_ca_cert] = kafka_ca_cert if kafka_ca_cert
+      end
+    end
+
     def kafka
-      @kafka ||= Kafka.new(BROKERS, client_id: CLIENT_ID) if BROKERS.any?
+      @kafka ||= Kafka.new(self::BROKERS, **kafka_config) if self::BROKERS.any?
```

Comment:
Using `self::` here allows overriding the value in our tests. It should be a no-op for functionality.